src/storm/mesos/logviewer/LogViewerController.java

```diff
@@ -0,0 +1,211 @@
+package storm.mesos.logviewer;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.ProcessBuilder.Redirect;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.yaml.snakeyaml.Yaml;
+
+import com.google.common.base.Optional;
+
+import storm.mesos.logviewer.IUrlDetection;
+import storm.mesos.logviewer.SocketUrlDetection;
+
+public class LogViewerController implements ILogController {
+
+    /**
+     * Create a log view controller with a SocketUrlDetector set to 
+     * the port that is in the storm.yaml config file. If no port is
+     * set then use the default of 8000. 
+     */
+    public LogViewerController() {
+        // Use a simplistic approach.
+        this(new SocketUrlDetection());
+    }
+    
+    /**
+     * Load a preconfigured Url detector to check for an existing logviewer.
+     * 
+     * @param urlDetector
+     */
+    public LogViewerController(IUrlDetection urlDetector) {
+        setUrlDetector(urlDetector);
+        initialize();
+    }
+    
+    /**
+     * Start up the logviewer, but before that is done a check is made to
+     * see if an existing logviewer (or process) is on the same port. If so
+     * an attempt is made to close it down. If unsuccessful then do not start
+     * a logviewer.
+     * 
+     */
+    @Override
+    public void start() {
+       try {
+           if (exists()){
+               // The reason I would want to terminate is because:
+               // a. there is a conflict which should be resolved
+               // b. assuming this is a logviewer that was started by a now 
+               //    defunct supervisor its probably pointing at the wrong log dir.
+               //
+               terminateExistingViewer();
+               launchLogViewer();
+           }
+           else {
+               launchLogViewer();
+           }    
+       } catch (Exception e) {
+           LOG.error("Failed to start logviewer", e);
+       }
+    }
+    
+    @Override
+    public void stop() {
+        getProcess().destroy();
+    }
+    
+    @Override
+    public boolean exists() {
+        return getUrlDetector().isReachable();
+    }
+    
+    @Override
+    public void terminateExistingViewer() {
+        LOG.info("Terminating existing viewer...."); 
```

Comment:
I don't quite understand, are you hoping to merge this as is? This doesn't look like it will ever work ?